site/content/guides/setup-scripts.md

```diff
@@ -0,0 +1,145 @@
+
+```js
+const { data } = await axios({ // create dummy check on Checkly
+  ...
+  data: {
+    name: "dummy_check",
+    checkType: "BROWSER",
+    frequency: 5,
+    activated: true,
+    locations: ["us-east-1"],
+    script:
+      'const assert = require("chai").assert;const puppeteer = require("puppeteer");const browser = await puppeteer.launch();const page =await browser.newPage();await page.goto("https://google.com/");const title = await page.title();assert.equal(title, "Google"); await new Promise(resolve => setTimeout(resolve, 1000)); browser.close();',
+    useGlobalAlertSettings: true,
+    degradedResponseTime: 10000,
+    maxResponseTime: 20000
+  }
+}
+```
+
+The last bit of the script is where we will extract the `id` of the newly created dummy check from the response to our axios request, in order to pass it on to our API check's own HTTP request. Specifically, we will adding the `id` to the target URL, as the target endpoint (`DELETE /v1/checks/{id}`) requires.
+
+```js
+const checkId = data.id; // extract dummy check id from response
+request.url = request.url + "/" + checkId; // pass check id as path param in check's main http request url
+```
+
+Putting everything together, the final result is:
+
+```js
+const axios = require("axios"); // import axios library explicitly
+const apiKey = process.env.API_KEY;
+
+const { data } = await axios({ // create dummy check on Checkly
+  method: "post",
+  url: "https://api.checklyhq.com/v1/checks",
+  headers: {
+    Authorization: `Bearer ${apiKey}`
+  },
+  data: {
+    name: "dummy_check",
+    checkType: "BROWSER",
+    frequency: 5,
+    activated: true,
+    locations: ["us-east-1"],
+    script:
+      'const assert = require("chai").assert;const puppeteer = require("puppeteer");const browser = await puppeteer.launch();const page =await browser.newPage();await page.goto("https://google.com/");const title = await page.title();assert.equal(title, "Google"); await new Promise(resolve => setTimeout(resolve, 1000)); browser.close();',
+    useGlobalAlertSettings: true,
+    degradedResponseTime: 10000,
+    maxResponseTime: 20000,
+  }
+})
+
+const checkId = data.id; // extract dummy check id from response
+request.url = request.url + "/" + checkId; // pass check id as path param in check's main http request url
+```
+
+We have built a fully autonomous and self-contained API check, which will take care of setting up all it needs to verify our endpoint is working correctly, then proceed to do just that.
+
+Note that the check might still fail due to an issue with an endpoint other than the one we are checking (`DELETE /v1/checks/{id}`), as the `POST /v1/checks` call in our setup script might also break at some point. Here, the Checkly check result will let us know whether the error occurred in the setup script or in the main HTTP request, allowing us to easily recognise the cause of the issue.
```

Comment:
Maybe?
take care of setting = set
to easily recognise the cause of the issue = to recognise the cause of the issue easily.